src/NzbDrone.Core/Music/AddAlbumService.cs

```diff
@@ -20,40 +21,93 @@ public interface IAddAlbumService
     public class AddAlbumService : IAddAlbumService
     {
         private readonly IAlbumService _albumService;
+        private readonly IReleaseService _releaseService;
         private readonly IProvideAlbumInfo _albumInfo;
         private readonly IArtistMetadataRepository _artistMetadataRepository;
         private readonly IRefreshTrackService _refreshTrackService;
         private readonly Logger _logger;
 
         public AddAlbumService(IAlbumService albumService,
+                               IReleaseService releaseService,
                                IProvideAlbumInfo albumInfo,
                                IArtistMetadataRepository artistMetadataRepository,
                                IRefreshTrackService refreshTrackService,
                                Logger logger)
         {
             _albumService = albumService;
+            _releaseService = releaseService;
             _albumInfo = albumInfo;
             _artistMetadataRepository = artistMetadataRepository;
             _refreshTrackService = refreshTrackService;
             _logger = logger;
         }
 
-        public Album AddAlbum(Album newAlbum)
+        public List<AlbumRelease> AddAlbumReleases(Album album)
         {
-            Ensure.That(newAlbum, () => newAlbum).IsNotNull();
-
-            var tuple = AddSkyhookData(newAlbum);
-            newAlbum = tuple.Item2;
+            var remoteReleases = album.AlbumReleases.Value.DistinctBy(m => m.ForeignReleaseId).ToList();
+            var existingReleases = _releaseService.GetReleasesForRefresh(album.Id, remoteReleases.Select(x => x.ForeignReleaseId));
```

Comment:
Are there ever going to be existing releases for a New Album? Isn't the Album>Release relationship a 1:Many?

Reply:
Thinking back thru I assume this is for when a Album MBID changes, thus Lidarr thinks its a New Album, but the releases are already there....